src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+        /// <param name="shuntResistance">The value of the shunt resistor. This defaults to 0.1 Ohms as used in the Adafruit breakout board.</param>
+        public void SetCalibration(PgaSensitivity sensitivity, float fsCurrent, float shuntResistance = 0.1F)
+        {
+            int i;
+            float minCurrentLsb = fsCurrent / 32767;
+            float maxCurrentLsb = fsCurrent / 4096;
+            float currentStep;
+
+            if (fsCurrent > _sensitivityVoltage[sensitivity] / shuntResistance)
+            {
+                throw new ArgumentOutOfRangeException(nameof(fsCurrent), "Current range too large for given shunt resistance and amplifier sensitivity.");
+            }
+
+            // find a step by taking a value in the same range as the maxCurrentLSB
+            currentStep = (float)Math.Pow(10, (int)Math.Log10(maxCurrentLsb) - 1);
+
+            // step through rounded values of currentLsb to find the best LSB for the current.
+            i = 1;
+            while (i * currentStep < minCurrentLsb) { i++; };
+            CurrentLsb = i * currentStep;
+            PowerLsb = CurrentLsb * 20;
+
+            SetPGARange(sensitivity);
+            SetCalibration((ushort)(0.04096F / (CurrentLsb * shuntResistance)));
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposeI2cDevice & disposing)
+            {
+                _i2cDevice?.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Dispose of managed assets.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        /// <summary>
+        /// Get the measured shunt voltage.
+        /// </summary>
+        /// <returns>The shunt voltage in Volts</returns>
+        public float GetShuntVoltage()
+        {
+            // read the shunt voltage. LSB = 10uV then convert to Volts
+            return (short)ReadRegister(Register.ShuntVoltage, _readDelays[(AdcResolutionOrSamples)_shuntAdcResSamp]) * 10 / 1000000F;
+        }
+
+        /// <summary>
+        /// Get the measured Bus voltage.
+        /// </summary>
+        /// <returns>The Bus voltage in Volts</returns>
+        public float GetBusVoltage()
+        {
+            // read the bus voltage. LSB = 4mV then convert to Volts
+            return ((short)ReadRegister(Register.BusVoltage, _readDelays[_busAdcResSamp]) >> 3) * 4 / 1000F;
+        }
+
+        /// <summary>
+        /// Get the calculated current through the INA219.
+        /// </summary>
+        /// <remarks>
+        /// This value is determined by an internal calculation using the calibration register and the read shunt voltage and then scaled
+        /// using the CurrentLsb property.
+        /// <see cref="CurrentLsb"/>
+        /// </remarks>
+        /// <returns>The calculated current in Amperes</returns>
+        public float GetCurrent()
```

Comment:
I think this should be called `ReadCurrent`, property should be removed and `SetCalibration` should be private and likely renamed (considering context this is used at I doubt this is calibrating anything - it's most likely adjusting output to human readable units)

Reply:
Here is a good example of method. This is doing an actual operation which might take some longer time in some cases (i.e. sensor is still initializing etc). In other cases you're just changing internal registers. This case also even if timing is small returns potentially different value each time it's called without any state change and this is 100% case where we should use method